pkg/controller/components/apiproxy.go

```diff
@@ -0,0 +1,166 @@
+package components
+
+import (
+	"fmt"
+
+	networkingv1 "k8s.io/api/networking/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/dominodatalab/distributed-compute-operator/pkg/cluster/metadata"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/actions"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
+)
+
+const (
+	defaultAPIProxyPort        = 8899
+	targetServicePort          = 8899
+	apiProxyPortName           = "api-proxy"
+	component                  = "api-proxy"
+	executionIDLabel           = "dominodatalab.com/execution-id"
+	projectIDLabel             = "dominodatalab.com/project-id"
+	datasourceProxyClientLabel = "datasource-proxy-client"
+)
+
+func executionID(obj *client.Object) string {
+	if executionID, ok := (*obj).GetLabels()[executionIDLabel]; ok {
+		return executionID
+	}
+	return "unknown-execution-id" // primarily, for integration testing
+}
+
+func runPodSelector(obj *client.Object) map[string]string {
```

Comment:
This and `executionID` should probably not be in DCO. selectors should be supplied by the client. I believe we may already have this: https://github.com/dominodatalab/distributed-compute-operator/blob/8a0b61620d2e8b81ca37d2214d8ce386e6473c84/api/v1alpha1/common_types.go#L48

Reply:
I will check `clientLabels`. On the other hand, why are they supplied by the client? By definition here, we set this service to work only with the current run pod, which we can probably determine internally. I suppose, there may be variations in the name if the run pod?

Reply:
The DCO doesn't really know / manage the Run Pod. There is just _some_ Pod which will connect to the cluster as a client. So Nucleus specifies the `labelSelector` for the client.

Reply:
OK, agree.